empire/runner.go

```diff
@@ -81,48 +83,42 @@ func (r *relayer) Relay(ctx context.Context, f *postContainersForm) (*ContainerR
 	return cr, nil
 }
 
-type runner struct {
-	store   *store
-	relayer containerRelayer
+type fakeRunner struct{}
+
+func (r *fakeRunner) Run(_ context.Context, opts runner.RunOpts) error {
+	return nil
+}
+
+type runnerService struct {
+	store  *store
+	runner interface {
+		Run(context.Context, runner.RunOpts) error
+	}
 }
 
-func (r *runner) Run(ctx context.Context, app *App, command string, opts ProcessesRunOpts) (*ContainerRelay, error) {
-	c, err := r.newContainerForm(ctx, app, command, opts)
+func (r *runnerService) Run(ctx context.Context, app *App, command string, in io.Reader, out io.Writer) error {
+	opts, err := r.newRunOpts(ctx, app, command)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	return r.relayer.Relay(ctx, c)
+	opts.Input = in
+	opts.Output = out
+
+	return r.runner.Run(ctx, opts)
```

Comment:
It might be good to actually put the `Run` method behind the `service.Manager` interface, since ideally anything that needs to be run should go through the manager.